### PR TITLE
feat: allow for hoverDimension to be configured

### PR DESCRIPTION
**Pull request by rawkode**
This PR allows for the user to configure the `hoverDimension` within line graphs, so they can choose between the ability to see all series values at a time range; or the ability to select and highlight a single series.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by ebb-tide** [comment]
Thanks so much @rawkode ! Could you add a screenshot of this in action please?

File: ui/src/timeMachine/components/view_options/LineOptions.tsx [hunk 1]
```diff
@@ -58,6 +59,7 @@ interface OwnProps {
   geom?: XYGeom
   colors: Color[]
   shadeBelow?: boolean
+  hoverDimension?: 'auto' | 'x' | 'y' | 'xy'
```
Comment (ebb-tide) [inline ui/src/timeMachine/components/view_options/LineOptions.tsx:62]: hoverDimension is an optional argument, but I don't think the dropdown will have anything selected when it is not provided. I think this component should handle setting hoverDimension to 'auto' if none have been provided.

**Comment by rawkode** [comment]
@ebb-tide I've uploaded a small video

https://streamable.com/5rjqdy

**Comment by rawkode** [comment]
CC @jsteenb2, as I guess `pkger` will also need updated

**Comment by rawkode** [comment]
Added fixes required for pkger to work too

File: pkger/parser_models.go [hunk 2]
```diff
@@ -495,31 +495,32 @@ func (d *dashboard) valid() []validationErr {
...
+	fieldChartDomain         = "domain"
+	fieldChartFillColumns    = "fillColumns"
+	fieldChartGeom           = "geom"
+	fieldChartHeight         = "height"
+	fieldChartLegend         = "legend"
+	fieldChartNote           = "note"
+	fieldChartNoteOnEmpty    = "noteOnEmpty"
+	fieldChartPosition       = "position"
+	fieldChartQueries        = "queries"
+	fieldChartShade          = "shade"
```
Comment (jsteenb2) [inline pkger/parser_models.go:512]: very excited to be able to automate this entire file out of existence 🙃

File: pkger/testdata/dashboard_single_stat_plus_line.json [hunk 3]
```diff
@@ -69,4 +70,4 @@
       ]
     }
   }
-]
+]
```
Comment (jsteenb2) [inline pkger/testdata/dashboard_single_stat_plus_line.json:73]: oh no, my newline 😢

File: pkger/testdata/dashboard_xy.json [hunk 4]
```diff
@@ -60,4 +61,4 @@
       ]
     }
   }
-]
+]
```
Comment (jsteenb2) [inline pkger/testdata/dashboard_xy.json:64]: bring the newline back plz 🙏

File: pkger/parser_models.go [hunk 5]
```diff
@@ -657,6 +659,7 @@ func (c chart) properties() influxdb.ViewProperties {
 			XColumn:           c.XCol,
 			YColumn:           c.YCol,
 			ShadeBelow:        c.Shade,
+			HoverDimension:    c.HoverDimension,
```
Comment (jsteenb2) [inline pkger/parser_models.go:662]: to confirm, the hover dimension is only valid for single stat plus line and the xy views?

File: pkger/testdata/dashboard_single_stat_plus_line.json (hunk 3 above)
Comment (rawkode) [reply pkger/testdata/dashboard_single_stat_plus_line.json:73]: Ack, my IDE obeys the `.editorconfig` and the InfluxDB is rather lacking; so they're purged :'(

File: pkger/parser_models.go (hunk 5 above)
Comment (rawkode) [reply pkger/parser_models.go:662]: Correct

**Comment by rawkode** [comment]
Did we do something that stopped the tests running?

**Comment by jsteenb2** [comment]
thats odd 😳

**Comment by rawkode** [comment]
Will merge once these tests pass

**Pull request merged**